artemis-server/src/main/java/org/apache/activemq/artemis/core/settings/impl/HierarchicalObjectRepository.java

```diff
@@ -488,12 +488,11 @@ public int compare(final String o1, final String o2) {
             String[] rightSplits = o2.split(quotedDelimiter);
             for (int i = 0; i < leftSplits.length; i++) {
                String left = leftSplits[i];
-               if (left.equals(singleWord)) {
-                  if (rightSplits.length < i || !rightSplits[i].equals(singleWord)) {
-                     return -1;
-                  } else {
-                     return +1;
-                  }
+               String right = rightSplits[i];
```

Comment:
Now, the behavior LGTM but the `leftSplits.length` could be greater than `rightSplits.length` causing an ArrayIndexOutOfBoundsException, ie o1=`foo.*` and o2=`foo.*.too`

Reply:
i tried to reproduce the case but i guess the `getPossibleMatches` method ensures that the matches has the same length if it falls on the last if (both singleWord).
anyways i added an extra if to return in case the 2nd is shorter.